src/Umbraco.Cms.Api.Management/Controllers/Document/AllowedChildrenByKeyDocumentController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Asp.Versioning;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Api.Common.ViewModels.Pagination;
+using Umbraco.Cms.Api.Management.ViewModels.DocumentType;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Core.Services.OperationStatus;
+
+namespace Umbraco.Cms.Api.Management.Controllers.Document;
+
+public class AllowedChildrenByKeyDocumentController : DocumentControllerBase
+{
+    private readonly IContentTypeService _contentTypeService;
+    private readonly IUmbracoMapper _umbracoMapper;
+
+    public AllowedChildrenByKeyDocumentController(IContentTypeService contentTypeService, IUmbracoMapper umbracoMapper)
+    {
+        _contentTypeService = contentTypeService;
+        _umbracoMapper = umbracoMapper;
+    }
+
+    [HttpGet("{id:guid}/allowed-document-types")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(PagedViewModel<DocumentTypeResponseModel>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> AllowedChildrenByKey(Guid id, int skip = 0, int take = 100)
```

Comment:
This id is a content/document id.. and not a content type id. Service method should most likely be located on a new `IContent*Service` like we have the `IContentEditingService.`

Reply:
From a service layer perspective it makes good sense to keep the current services, but we need one on a IContent*Service that look up the content type from a content id, and calls these methods

Reply:
I've tried creating a `IContentCreatingService` (naming is hard, might need a new name) instead 👍 
It uses the contentService to get content out, and then calls the new `GetAllowedChildrenAsync` on the IContentTypeService